processor/aggregationprocessor/README.md

```diff
@@ -0,0 +1,141 @@
+
+## Minimum collector versions
+- Introduced: [v1.18.0](https://github.com/observIQ/observiq-otel-collector/releases/tag/v1.18.0)
+
+## Supported pipelines
+- Metrics
+
+## How it works
+1. The user configures the aggregation processor in the desired metrics pipeline.
+2. Every metric that flows through the pipeline is matched against the provided `include` regex.
+3. If the metric name does not match the `include` regex, the metric passes through the processor.
+4. If the metric matches, but is not a gauge or cumulative sum, the metric passes through the processor.
+5. If the metric name does match, and the metric is a gauge or cumulative sum, the metric is added to an aggregate based on its attributes. The metric does not continue down the pipeline.
+6. After the configured `interval` has passed, all aggregate metrics are emitted. Aggregate metrics are emitted with a name based on the configured `metric_name` for the aggregation.
+7. All aggregations are cleared, and will not be emitted on the next interval, unless another matching metric enters the pipeline.
+
+## Configuration
+| Field               | Type   | Default | Description                                                            |
+|---------------------|--------|---------|------------------------------------------------------------------------|
+| `interval` | duration | `1m` | The interval on which to emit aggregate metrics. |
+| `include` | regex | `"^.*$"` | A regex that specifies which metrics to consider for aggregation. The default regex matches all metrics. |
+| `aggregations` | []map | | A list of aggregations to perform on each metric.|
+
+
+<table>
+<tr><th>Field</th><th>Type</th><th>Default</th><th>Description</th></tr>
+
+<tr><td><pre>interval</pre></td><td>duration</td><td><pre>1m</pre></td><td>The interval on which to emit aggregate metrics. </td></tr>
+
+<tr><td><pre>include</pre></td><td>regular expression</td><td><pre>"^.*$"</pre></td><td>A regex that specifies which metrics to consider for aggregation. The default regex matches all metrics. </td></tr>
+
+<tr><td><pre>aggregations</pre></td><td>[]map</td><td><pre>
+- type: min
+  metric_name: "$$0.min"
+- type: max
+  metric_name: "$$0.max"
+- type: avg
+  metric_name: "$$0.avg"
+</pre></td><td>A list of aggregations to perform on each metric. By default, min, max, and average aggregate metrics are emitted every `interval`.</td></tr>
+
+<tr><td><pre>aggregations[].type</pre></td><td>aggregate type</td><td></td><td> The type of the aggregation. Valid values are: `min`, `max`, `avg`, `first`, `last`.</td></tr>
+
+<tr><td><pre>aggregations[].metric_name</pre></td><td>string</td><td><pre>"$$0"</pre></td><td>The name of the metric emitted for this aggregation. By default, the portion of the name matched by the `include` regex is used.</td></tr>
+</table>
```

Comment:
I made this an html table because of the multi line default for the `aggregations` section; Don't know if we'd prefer this or if we'd prefer doing  something like the mask processor: https://github.com/observIQ/observiq-otel-collector/tree/main/processor/maskprocessor#configuration - I personally would like for the default here to be copy/paste-able

Reply:
Personally to me the maskprocessor is easier to read without having to know the html syntax.

Reply:
It's definitely annoying from a dev POV. I'll try changing it over to the MD syntax, but the problem is that the default for `aggregations` will probably not be easily copy-pasteable, which probably sucks if you just want to tweak the default config for it from a user POV.

Reply:
Actually, I think maybe I found a good middle-ground? Not sure how I feel about it, but it is copy-pasteable valid yaml in the MD format here.

https://github.com/observIQ/observiq-otel-collector/blob/eb9f373bae9780978bade70592bd3495964f292f/processor/aggregationprocessor/README.md

Reply:
looks good to me